src/PufferVaultV2.sol

```diff
@@ -444,6 +475,24 @@ contract PufferVaultV2 is PufferVault, IPufferVaultV2 {
         $.exitFeeBasisPoints = newExitFeeBasisPoints;
     }
 
+    // slither-disable-next-line dead-code
+    modifier markDeposit() virtual {
+        assembly {
+            tstore(_DEPOSIT_TRACKER_LOCATION, 1)
+        }
+        _;
+    }
+
+    modifier revertIfDeposited() virtual {
+        assembly {
+            if tload(_DEPOSIT_TRACKER_LOCATION) {
+                mstore(0x00, 0x39b79d11) // `DepositAndWithdrawalForbidden()`.
```

Comment:
Shouldn't this be `tstore` to use transient storage rather than the memory?

Reply:
and please document what is being stored `0x39b79d11`

Reply:
On second look, I noticed  `_DEPOSIT_TRACKER_LOCATION` is what is is being stored on transient storage, so forget my first comment. 

just for the second comment, we possibly can just use the error type with the proper name right?

Reply:
We can't use the normal revert syntax in the assembly block. We could make it uglier:
```solidity
    modifier revertIfDeposited() virtual {
        uint256 tracker;
        assembly {
            tracker := tload(_DEPOSIT_TRACKER_LOCATION)
        }
        if (tracker != 0) {
            revert DepositAndWithdrawalForbidden();
        }
        _;
    }
```
So I just added more comments to the existing version.